packages/client/src/components/Location/LocationFinder.tsx

```diff
@@ -63,9 +63,20 @@ const LocationFinder: React.FC<LocationFinderProps & MatomoTrackEventProps> = ({
   const [autoSuggestValue, setAutoSuggestValue] = useState<string>("");
   const [touched, setTouched] = useState<{ [key: string]: boolean }>({});
 
+  const splitHouseNumber = houseNumberFull
+    ?.toUpperCase()
+    ?.match(/[a-z]+|[^a-z]+/gi);
+
+  const graphQlHouseNumber = splitHouseNumber
+    ?.map((part) => `${part.trim()} `) // Split each part with a space
+    .toString() // Revert to one string
+    .replace(/,/g, "") // Remove all commas
+    .replace(/\s+/g, " ") // Remove duplicate spaces
+    .trim(); // Remove and spaces
```

Comment:
This is still very messy. Possible cleanups:
- I see who `trim()` functions, is it not possible to use only one?
- I see two `replace()` functions and one `toString()`, can you not combine into one regex?
- I've created the util `stripString`, can you not use that on each `part`? That also lowercases and trims